### PR TITLE
add `str escape-glob` command

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description
This pr is a follow up to #11621, it introduces a `str escape-glob` command as a workaround for the case:

```nushell
let f = "a[123]b"
ls $f
```

It will glob `a[123]b`, we can get rid of the behavior through `str escape-glob` command:

```nushll
let f = "a[123]b"
ls ($f | str escape-glob)
```

It's more useful in the `each` context:
`ls | get name | str escape-glob | each {|it| ls $it}`

# User-Facing Changes
NaN

# Tests + Formatting
Done

# After Submitting

### CODE REVIEW

File: crates/nu-command/src/strings/str_/escape_glob.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        let cell_paths: Vec<CellPath> = call.rest(engine_state, stack, 0)?;
+        let cell_paths = (!cell_paths.is_empty()).then_some(cell_paths);
+        let args = Arguments { cell_paths };
+        operate(action, args, input, call.head, engine_state.ctrlc.clone())
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "escape glob pattern before list",
+            example: r#"let f = test[a]; ls ($f | str escape-glob)"#,
```
Comment (fdncred) [inline crates/nu-command/src/strings/str_/escape_glob.rs:71]: I think this is invalid without quotes.
```suggestion
            example: r#"let f = 'test[a]'; ls ($f | str escape-glob)"#,
```

**Review by fdncred — APPROVED** [review]
Except for the one minor thing in the example, I think this is an interesting approach.

**Comment by fdncred** [comment]
This kind of works
```nushell
ls | get name | str escape-glob | each {|it| ls -D $it} | flatten
```
but this shows a different result.
```nushell
❯ touch "test[a]"
❯ ls | $in.name | str escape-glob
```
I'm not sure if that's intended or not?

**Comment by WindSoilder** [comment]
I think it's intended, it shows escaped result.  And that's why it works in the first case you provided.

**Pull request merged**